src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1867,488 +2163,167 @@ int XrdHttpReq::PostProcessHTTPReq(bool final_) {
-          if (prot->SecEntity.vorg) {
-            stringresp += " VO: ";
-            stringresp += prot->SecEntity.vorg;
-          }
-          
-          if (prot->SecEntity.moninfo) {
-            stringresp += " DN: ";
-            stringresp += prot->SecEntity.moninfo;
-          } else
-            if (prot->SecEntity.name) {
-              stringresp += " DN: ";
-              stringresp += prot->SecEntity.name;
-            }
-          
-          
-          if (prot->SecEntity.role) {
-            stringresp += " Role: ";
-            stringresp += prot->SecEntity.role;
-            if (prot->SecEntity.endorsements) {
-              stringresp += " (";
-              stringresp += prot->SecEntity.endorsements;
-              stringresp += ") ";
-            }
-          }
-          
-          
-           
-          if (prot->SecEntity.vorg ||
-            prot->SecEntity.moninfo ||
-            prot->SecEntity.role)
-            stringresp += " )";
-          
-          if (prot->SecEntity.host) {
-            stringresp += " ( ";
-            stringresp += prot->SecEntity.host;
-            stringresp += " )";
+        case 2: // close file handle in case of the directory
+        {
+          if (xrdresp != kXR_ok) {
+            prot->SendSimpleResp(httpStatusCode, NULL, NULL,
+                                  httpStatusText.c_str(), httpStatusText.length(), false);
+            return -1;
           }
-          
-          stringresp += "</span></p>\n";
-          stringresp += "<p>Powered by XrdHTTP ";
-          stringresp += XrdVSTRING;
-          stringresp += " (CERN IT-SDC)</p>\n";
-          
-          prot->SendSimpleResp(200, NULL, NULL, (char *) stringresp.c_str(), 0, keepalive);
-          stringresp.clear();
-          return keepalive ? 1 : -1;
+          return 0;
         }
-
-
-      } // end handling of dirlist
-      else
-      { // begin handling of open-read-close
-
-        // To duplicate the state diagram from the rtGET request state
-        // - 0: Perform a stat on the resource
-        // - 1: Perform a checksum request on the resource (only if requested in header; otherwise skipped)
-        // - 2: Perform an open request (dirlist as appropriate).
-        // - 3+: Reads from file; if at end, perform a close.
-        switch (reqstate) {
-          case 0: //stat
+        case 3: // handle the directory listing response
+        {
+          return PostProcessListing(final_);
+        }
+        default: //read or readv, followed by a close.
+        {
+          // If we are postprocessing a close, potentially send out informational trailers
+          if ((ntohs(xrdreq.header.requestid) == kXR_close) || readClosing)
           {
-            // Ugly hack. Be careful with EOS! Test with vanilla XrdHTTP and EOS, separately
-            // A 404 on the preliminary stat() is fatal only
-            // in a manager. A non-manager will ignore the result and try anyway to open the file
```

Comment:
@ccaffy I wonder if the changes introduced here will affect EOS, do you know? If that's possible, we should test this with EOS to make sure nothing there is relying on the previous behavior of stat + (open | dirlist).

Reply:
I would indeed appreciate a test from the EOS side as well!